fairroot/base/source/FairFileSource.cxx

```diff
@@ -73,6 +75,7 @@ FairFileSource::FairFileSource(TFile* f, const char* Title, UInt_t)
     , fEventMeanTime(0.)
     , fCheckFileLayout(kTRUE)
 {
+    fRootFile = maybe_owning_ptr<TFile>{f, non_owning};
```

Comment:
Hm, why assigning here instead of initializing in the initializer list?

Reply:
Because it's a member variable of the parent class?

We could change the ctor of the parent class to allow passing a maybe_ptr in…

Reply:
Yes pls, and then you can also deduplicate the `if ((!fRootFile) || fRootFile->IsZombie())`, no?

Reply:
done.